src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -752,7 +751,7 @@ public static int onBlockBreakEvent(World world, GameType gameType, EntityPlayer
     {
         // Logic from tryHarvestBlock for pre-canceling the event
         boolean preCancelEvent = false;
-        if (gameType.isCreative() && !entityPlayer.getHeldItemMainhand().isEmpty() && entityPlayer.getHeldItemMainhand().getItem() instanceof ItemSword)
+        if (gameType.isCreative() && !entityPlayer.getHeldItemMainhand().isEmpty() && !entityPlayer.getHeldItemMainhand().getItem().canDestroyBlockInCreative(entityPlayer.getHeldItemMainhand(), world.getBlockState(pos), entityPlayer))
```

Comment:
That makes sense. There's no line restriction here so it should be made readable.